Services/CommentRenderService.cs

```diff
@@ -0,0 +1,127 @@
+﻿/*
+Copyright 2021, E.J. Wilburn, Marcus McKinnon, Kevin Williams
+This program is distributed under the terms of the GNU General Public License.
+
+This file is part of Palaver.
+
+Palaver is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Palaver is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Palaver.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+namespace PalaverCore.Services;
+
+using Markdig;
+using Markdig.Prism;
+using System;
+using System.Text.RegularExpressions;
+using static PalaverCore.Models.Comment;
+
+public class CommentRenderService
+{
+    private MarkdownPipeline _pipeline;
+
+    public CommentRenderService()
+    {
+        _pipeline = new MarkdownPipelineBuilder()
+            .UseAdvancedExtensions()
+            .UsePrism()
+            .UseSoftlineBreakAsHardlineBreak()
+            .Build();
+    }
+
+    /// <summary>
+    /// Apply temporary filtering for comment text display, such as modifying images so they're lazy loaded.
+    /// </summary>
+    /// <param name="commentText"></param>
+    /// <param name="format"></param>
+    /// <returns></returns>
+    public string ToHtml(string commentText, TextFormat format)
+    {
+        if (String.IsNullOrWhiteSpace(commentText))
+            return commentText;
+
+        string output = commentText;
+        switch (format)
+        {
+            case TextFormat.Markdown:
+                output = Markdown.ToHtml(output, _pipeline);
+                break;
+        }
```

Comment:
This code and the constructor are the only real new things in this class, it was moved out of the Comment class.